src/bin/sol-fbp-runner/runner.c

```diff
@@ -73,18 +74,22 @@ search_fbp_file(struct runner *r, const char *basename)
     uint16_t i;
     int err;
 
-    SOL_PTR_VECTOR_FOREACH_IDX (r->fbp_paths, p, i) {
-        err = asprintf(&fullpath, "%s/%s", p, basename);
-        if (err < 0) {
-            errno = ENOMEM;
-            return NULL;
-        }
-
-        if (stat(fullpath, &s) == 0 && S_ISREG(s.st_mode))
-            return fullpath;
-
-        free(fullpath);
-    }
+#define STAT_PATH(_path) \
+    do { \
+        err = asprintf(&fullpath, "%s/%s", _path, basename); \
+        if (err < 0) { \
+            errno = ENOMEM;\
+            return NULL; \
+        } \
+        if (stat(fullpath, &s) == 0 && S_ISREG(s.st_mode)) \
+            return fullpath; \
+        free(fullpath); \
+    } while (0)
```

Comment:
we are using macros like this in other places, is there a good advantage making this a funcion ?

I do not see any problem in change it, i'm just afraid i've missed something